zarr/util.py

```diff
@@ -670,7 +670,7 @@ def all_equal(value: Any, array: Any):
         # optimized to return on the first truthy value in `array`.
         try:
             return not np.any(array)
-        except TypeError:  # pragma: no cover
+        except (TypeError, ValueError):  # pragma: no cover
```

Comment:
What causes the `ValueError`? Is there a particular array type or value that `np.any` is raising on?

Reply:
It occurs when a category value (`baz`) gets passed to `int()`.

Reply:
Interesting. Looks like the error message and type changed from NumPy 1.21 to 1.22. Don't think that was intentional, but could be wrong. Raised issue ( https://github.com/numpy/numpy/issues/20898 )

In any event this workaround seems reasonable in the interim

Reply:
Great, thanks! Once this is green, I'll move forward with 2.11.0.